spec/matchers/discussion_matchers.rb

```diff
@@ -0,0 +1,96 @@
+require "rspec/expectations"
+
+module DiscussionMatchers
+  extend RSpec::Matchers::DSL
+
+  matcher :have_correct_discussion_title_on_page do |discussion_title|
+    match do |actual|
+      expect(actual).to have_text(discussion_title)
+    end
+    failure_message do |actual|
+      "expected that the discussion title '#{discussion_title}' would be present on the page."
+    end
+  end
+
+  matcher :have_correct_discussion_body_on_page do |discussion_body|
+    match do |actual|
+      expect(actual).to have_text(discussion_body)
+    end
+    failure_message do |actual|
+      "expected that the discussion body '#{discussion_body}' would be present on the page."
+    end
+  end
+
+  matcher :have_correct_discussion_message_for_action do |action|
+    expected_text = "Discussion was successfully #{action}."
+    match do |actual|
+      expect(actual).to have_text(expected_text)
+    end
+    failure_message do |actual|
+      "expected that the message '#{expected_text}' would be present on the page for the appropriate action."
+    end
+  end
+
+  matcher :have_expected_discussion_buttons do
+    match do
+      find_button "Ban"
+      find_button "Pin"
+      find_button "Lock"
+      find_button "Edit"
+      find_button "Delete"
+    end
+    failure_message do
+      "expected that the buttons `Edit` and `Create Discussion` exist on the page."
+    end
+  end
+
+  matcher :have_write_comment_section do
+    match do |actual|
+      expect(actual).to have_text("Write a Comment")
+      expect(actual).to have_text("Write")
+      expect(actual).to have_text("Preview")
+      expect(actual).to have_text("Body")
+      find_button "Submit"
+    end
+  end
+
+  matcher :have_correct_user_info do
+    match do |actual|
+      expect(actual).to have_text("Test Person")
+      expect(actual).to have_text("Posts: 1")
+      find_button "Ban"
```

Comment:
This assumes that the logged in user always has banning privileges, which reduces the usefulness of this matcher to just moderators and admins, which no ability to glean that from the matcher.  Perhaps this is best suited being an explicit call inside the test when it makes sense rather than baking this into the custom matcher?